REGRESSION(285426@main): WebProcessProxyMessages.h: error: unknown type name 'SystemMemoryPressureStatus'

fujii opened the pull request:
#### 2f5f866715c5f6f9adff6c4aad93d83bf1b999bd
<pre>
REGRESSION(285426@main): WebProcessProxyMessages.h: error: unknown type name &apos;SystemMemoryPressureStatus&apos;
<a href="https://bugs.webkit.org/show_bug.cgi?id=281790">https://bugs.webkit.org/show_bug.cgi?id=281790</a>

Reviewed by Chris Dumez.

Windows and PlayStation ports were broken after
&lt;<a href="https://commits.webkit.org/285426@main">https://commits.webkit.org/285426@main</a>&gt;.

&gt; WebProcessProxyMessages.h(435,34): error: unknown type name &apos;SystemMemoryPressureStatus&apos;; did you mean &apos;WTF::SystemMemoryPressureStatus&apos;?
&gt;   435 |     using Arguments = std::tuple&lt;SystemMemoryPressureStatus&gt;;
&gt;       |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
&gt;       |                                  WTF::SystemMemoryPressureStatus
&gt; GeneratedSerializers.h(127,12): note: &apos;WTF::SystemMemoryPressureStatus&apos; declared here
&gt;   127 | enum class SystemMemoryPressureStatus : uint8_t;
&gt;       |            ^

&lt;wtf/SystemMemoryPressureStatus.h&gt; has to be included. It has &quot;using
WTF::SystemMemoryPressureStatus;&quot;.

* Source/WebKit/Scripts/webkit/messages.py:
(types_that_cannot_be_forward_declared):
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:

Canonical link: <a href="https://commits.webkit.org/285460@main">https://commits.webkit.org/285460@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/57bf2a010b0dda4e30aec8741de566ee4d6e5702

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/72729 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52154 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25528 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/76920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/23963 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59959 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/23763 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/76920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/23963 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75796 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47166 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62599 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/76920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/72236 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43814 "Passed tests") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/22292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/65668 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20424 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/78593 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16975 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/19549 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/65638 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/17023 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62607 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/64915 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/13220 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/6869 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11171 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47952 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/2739 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/49019 "Failed to checkout and rebase branch from PR 35473") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/50314 "Built successfully") | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/48764 "Failed to checkout and rebase branch from PR 35473") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/Scripts/webkit/messages.py:
```diff
@@ -449,6 +449,7 @@ def types_that_cannot_be_forward_declared():
         'PlatformXR::SessionMode',
         'PlatformXR::VisibilityState',
         'String',
+        'SystemMemoryPressureStatus',
```
Why do we have to do this? Why can't this enum be forward-declared? This usually works.

cdumez replied:
We don't need this for `WTFLogLevel` for example. I suspect there is a better way to fix this.

fujii replied:
`WTFLogLevel` is defined at the global scope. Please read the error message. All I need is this single line https://github.com/WebKit/WebKit/blob/77f02397ad6c9e34e93941a9967b2d6c21e4c1a7/Source/WTF/wtf/MemoryPressureHandler.h#L282

cdumez replied:
Please let me know if the build is still broken for you after https://commits.webkit.org/285454@main. It fixed the build for me (I was seeing the same error as you locally on Mac)

fujii replied:
Still same error message.

cdumez commented on Source/WebKit/UIProcess/WebProcessProxy.messages.in:
```diff
@@ -52,7 +52,7 @@ messages -> WebProcessProxy WantsDispatchMessage {
     DidReceiveBackgroundResponsivenessPing()
 
-    MemoryPressureStatusChanged(enum:uint8_t SystemMemoryPressureStatus status)
+    MemoryPressureStatusChanged(SystemMemoryPressureStatus status)
```
Trying alternative fix in https://github.com/WebKit/WebKit/pull/35474

fujii committed:
REGRESSION(285426@main): WebProcessProxyMessages.h: error: unknown type name 'SystemMemoryPressureStatus'
https://bugs.webkit.org/show_bug.cgi?id=281790

Reviewed by Chris Dumez.

Windows and PlayStation ports were broken after
<https://commits.webkit.org/285426@main>.

> WebProcessProxyMessages.h(435,34): error: unknown type name 'SystemMemoryPressureStatus'; did you mean 'WTF::SystemMemoryPressureStatus'?
>   435 |     using Arguments = std::tuple<SystemMemoryPressureStatus>;
>       |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
>       |                                  WTF::SystemMemoryPressureStatus
> GeneratedSerializers.h(127,12): note: 'WTF::SystemMemoryPressureStatus' declared here
>   127 | enum class SystemMemoryPressureStatus : uint8_t;
>       |            ^

<wtf/SystemMemoryPressureStatus.h> has to be included. It has "using
WTF::SystemMemoryPressureStatus;".

* Source/WebKit/Scripts/webkit/messages.py:
(types_that_cannot_be_forward_declared):
* Source/WebKit/UIProcess/WebProcessProxy.messages.in:

Canonical link: https://commits.webkit.org/285460@main
